crates/uniffi-bindgen-react-native/src/bindings/react_native/gen_typescript/templates/ErrorTemplate.ts

```diff
@@ -2,21 +2,29 @@
 
 {%- call ts::docstring(e, 0) %}
 export class {{ type_name }} extends Error {
-    private constructor(message: string) {
+    constructor(message: string) {
         super(message);
     }
     {%- if e.is_flat() %}
     {%-   for variant in e.variants() %}
-    {%-    call ts::docstring(variant, 4) %}
     {%-    let var_name = variant.name()|class_name(ci) %}
-    static {{ var_name }} = class {{ var_name }} extends {{ type_name }} {
-        constructor(message: string) { super(message); }
-    }
+    static {{ var_name }}: typeof _{{ type_name }}_{{ var_name }};
     {% endfor -%}
     {% else %}
     // non-flat errors aren't implemented yet.
     {%- endif %}
 }
+{%- if e.is_flat() %}
+{%-   for variant in e.variants() %}
+{%-    call ts::docstring(variant, 4) %}
+{%-    let var_name = variant.name()|class_name(ci) %}
+class _{{ type_name }}_{{ var_name }} extends {{ type_name }} {
+    constructor(message: string) { super(message); }
+}
+{{ type_name }}.{{ var_name }} = _{{ type_name }}_{{ var_name }};
```

Comment:
I bow to your superior typescript-fu: couple of questions:

1. does the docstring travel with the class definition to the static variable declaration? i.e. if type `OuterError.InnerVariant` and hover over `InnerVariant` in my editor, does it show the doc string?
2. would putting all these class definitions in an IIFE confuse the whatever was getting confused more, or would it be a valid way of not cluttering the namespace to avoid having the `_{{ type_name }}_` and just use the `{{ var_name }}`?

Reply:
I think you are correct, it should probably be on the static declarations. I will move that down.

As for question 2, it appeared that the Typescript compiler was unhappy about the number of anonymous inner classes.  The definitions that has a small number of errors worked, but those that had 10+ variants were exploding due to 'complexity'.  It seemed like by moving the class definition outside of the class (so they were no longer inner classes but proper global ones), it stopped being an issue.